src/prefect/utilities/logging.py

```diff
@@ -258,3 +258,34 @@ def get_logger(name: str = None) -> logging.Logger:
         return prefect_logger
     else:
         return prefect_logger.getChild(name)
+
+
+class RedirectToLog:
+    """
+    Custom redirect of stdout messages to logs
+
+    Args:
+        - logger (logging.Logger, optional): an optional logger to redirect stdout. If
+            not provided a logger names `stdout` will be created.
+    """
+
+    def __init__(self, logger: logging.Logger = None) -> None:
+        self.stdout_logger = logger or get_logger("stdout")
+
+    def write(self, s: str) -> None:
+        """
+        Write message from stdout to a prefect logger.
+        Note: blank newlines will not be logged.
+
+        Args:
+            s (str): the message from stdout to be logged
+        """
+        if s.strip():
+            self.stdout_logger.info(s)
```

Comment:
Apologies for submitting so many small reviews, but are we sure these belong as INFO logs vs. DEBUG?  I'd vote DEBUG but open to other ideas here

Reply:
I actually think INFO is more appropriate because the way I think about it is people use stdout to opt-in to printing output. Where as DEBUG to me generally seems as something that would be unintentional lower level automated logs

Reply:
@lauralorenz / @wagoodman want to be a tie-breaker?

Reply:
tl;dr I vote INFO

I agree with @joshmeek 's rationale. I do like DEBUG when I think of using `print()` only to quickly throw in little stuff early on in development assuming I will configure loggers later, but it seems the intention of this from the get-go is to provide a wholesale replacement for using the logging API even on deployed flows based on known user expectations.

In addition I think users would prioritize their personalized `print()` logs over stuff at debug level; I'm thinking of it in terms of it's better to a) make it easier to set logging levels to INFO to get rid of lower level debug logs *without* getting rid of their personalized `print()` logs, instead of b) making it easier to get rid of their own `print()` logs.

Reply:
It is decided!